src/Concerns/TransactionManager.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Jenssegers\Mongodb\Concerns;
+
+use Closure;
+use MongoDB\Driver\Exception\RuntimeException;
+use MongoDB\Driver\Session;
+use function MongoDB\with_transaction;
+
+trait TransactionManager
+{
+    /**
+     * A list of transaction session.
+     * @var Session|null
+     */
+    protected ?Session $session;
+
+    /**
+     * Get the existing session or null.
+     */
+    public function getSession(): ?Session
+    {
+        return $this->session ?? null;
```

Comment:
In this case, it is, as the typed property is not initialised (see https://3v4l.org/ct02o). This can be avoided by initialising the property with a `null` value in the declaration (see https://3v4l.org/NkjmT).

Reply:
Done